src/Modifier/Product/VariantAttributesModifier.php

```diff
@@ -153,12 +160,14 @@ public function apply(Type $product)
                 /** @var string|int|float $value */
                 $value = $attribute['value'];
 
+                $variantAttributes[$hash] = (string) $value;
+
                 if (in_array($hash, $integerAttributes)) {
-                    $variantAttributes[ $hash ] = (int) $value;
-                } elseif (in_array($hash, $floatAttributes)) {
-                    $variantAttributes[ $hash ] = (float) $value;
-                } else {
-                    $variantAttributes[ $hash ] = (string) $value;
+                    $variantAttributes[$hash] = (int) $value;
+                }
+
+                if (in_array($hash, $floatAttributes)) {
+                    $variantAttributes[$hash] = (float) $value;
```

Comment:
I think this block and the block above(line 134->141) is solving a same issue so I suggest you to create a function for this :smile: